java/lance-jni/src/blocking_dataset.rs

```diff
@@ -635,7 +635,8 @@ fn inner_create_index(
         | IndexType::Inverted
         | IndexType::NGram
         | IndexType::ZoneMap
-        | IndexType::BloomFilter => {
+        | IndexType::BloomFilter
+        | IndexType::Json => {
```

Comment:
We already have a scalar index named json.  I wonder if we can name this one FtsJson or InvertedJson?